### PR TITLE
Fix PR-37 external hcloud manager

### PR DESCRIPTION
**Pull request by mcadam**
Quick fix to just remove the extra flag `--cloud-provider=external` on provisioning external node.
Not fully tested right now as the build is broken, see #38

### CODE REVIEW

File: cmd/nodeUtil.go [hunk 1]
```diff
@@ -88,6 +88,8 @@ func (cluster *Cluster) CreateNodes(suffix string, template Node, datacenters []
 }
 
 func (cluster *Cluster) ProvisionNodes(nodes []Node) error {
+	nodes = cluster.Nodes
```
Comment (xetys) [inline cmd/nodeUtil.go]: this is not the right way, as the only time we need to provision all nodes, it's during creation. When adding new workers, this statement will cause unintended side-effects.

File: cmd/clusterAddExternalWorker.go [hunk 2]
```diff
@@ -142,6 +143,8 @@ An external server must meet the following requirements:
 
 		FatalOnError(err)
 
+		cluster.Nodes = append(cluster.Nodes, externalNode)
```
Comment (xetys) [inline cmd/clusterAddExternalWorker.go]: this statement already comes 4 lines later. I save the new state as soon as the first changing operation succeeded, as ProvisionNodes()

Comment (mcadam) [reply cmd/clusterAddExternalWorker.go]: Yep just realised how the wait mechanism works, it was not working :) was refactoring to add a proper function and wait for the installation to actually happen, maybe I dont need to anymore 👍

Comment (xetys) [reply cmd/clusterAddExternalWorker.go]: Yeah you just can remove this one

**Comment by mcadam** [comment]
Ok now we have the behaviour that we want as we don't add the kubelet option to specify external cloud, I also added the option to pass a SSH port on the external node add command. 
Also on the remove command we were not capturing the passphrase so it was not able to delete the node from kubectl directly.

**Comment by mcadam** [comment]
So after recreating a cluster from scratch and trying to add my eternal node, everything works as expected, kubelet is started without `--cloud-provider=external` but still the node gets deleted automatically, so not enough apparently.

**Comment by xetys** [comment]
This is because of the hcloud-controller-manager. 

Until the issue at that project, the only workaround is to delete that deployment. 

You could build a check, if the deployment is present and in that case, remove it before joining the new node

**Pull request closed without merging**